### ISSUE TITLE
Show big warning when `set-parent` wants to get a parent for *another branch*

### ISSUE THREAD

**Issue by ruudk** [git-town/git-town#4004]
### Steps to reproduce

First of all, happy Git Town Power User here 💙 

From time to time this is something that annoys me, because I don't expect it. 

I now know how to reproduce the bug.

Today, I did an online GitHub edit that created branch and PR. I decided to use `gh pr checkout ID` to pull in this branch locally, because it needed extra work that I did not expect initially.

This is a branch that was created outside of Git Town's knowledge.

Later, I started working on something unrelated. I created a new branch using `git town hack feature` and started working. Then after doing a bunch of commits, I decided to cherry-pick some of those commits to a separate branch. I used my [git hack-pick](https://github.com/git-town/git-town/issues/3667#issuecomment-2222804895) 💎 and proposed that feature on GitHub.

I then went back to my feature branch, where I cherry picked from, and ran `git town set-parent` to rebase that one on the cherry-picked one.

But this is where it happened. 

```
❯ git town set-parent
Selected parent branch for "online-github-pr": cherry-picked

Parent branch for feature

Please select the parent of branch "feature" or enter its number.
Most of the time this is the main branch (main).
``` 

Git Town prompted me to setup the parent for `online-github-pr` (the one I pulled via `gh` earlier). I did not think about this, and did not read the branch name, and just selected the parent (the cherry picked branch).

When I pressed enter, and Git Town asked me to do it again (now for the `feature` branch) I knew something was off.

## Possible solutions

* Git Town should always ask me to set the parent for the current branch first. It should *never* prompt it for a branch that the user does not expect. 
* Afterwards, it can prompt for those. But, if it does. It should make this super clear. Maybe with some colored warning. 

### Command output

```shell
See above.
```

**Comment by kevgo** [comment]
Thanks for reporting this issue. I was able to reproduce it on my end. Right now, the `set-parent` command insists on knowing the entire ancestry of all local branches. This leads to the situation you described: if there are any unrelated branches without known ancestry, the `git town set-parent` command asks for the parents of those unrelated branches first before asking for the new parent of the current branch. I agree that this behavior is super unintuitive.

A better approach would be to ignore the ancestry of other branches and just let the user enter the new parent. Then, if that new parent’s parent is unknown, we could ask for that parent.

**Comment by ruudk** [comment]
Thanks for fixing this! 🤩

**Comment by kevgo** [comment]
This is now available at part of [Git Town 16.2.1](https://github.com/git-town/git-town/releases/tag/v16.2.1).
